### ISSUE TITLE
Add Workspace Support

### ISSUE THREAD

**Issue by jbrinkman** [apideck-libraries/portman#118]
When using portman as part of my development workflow, I would like to have a new postman collection created if it does not already exist.  Currently, this is only supported for creating collections in My Workspace.  Even if I create an API key for a team workspace, the collection still only gets created in my private area.  In order to create a collection in a specific workspace, you need to provide a workspace id when creating the collection (as in the example below)

```typescript
    const queryUrl = workspaceId? `${this.baseUrl}/collections?workspace=${workspaceId}`: `${this.baseUrl}/collections`
    const config = {
      method: 'post',
      url: queryUrl,
      headers: {
        'Content-Type': 'application/json',
        'X-API-Key': this.apiKey,
      },
      data: data,
    } as AxiosRequestConfig;

    try {
      const res = await axios(config);
    } catch { ... }
```

As part of this feature, I would like to identify the workspace either by name or by Id.  Finding the specific workspace id is not easy in Postman and requires calling the Postman API, so having the option to identify the workspace by name is very useful.

**Comment by thim81** [comment]
@jbrinkman  This is an interesting feature addition, that we will definitely look into.
We will look into the Postman API to better understand the behaviour with or without the `workspace`.

At first sight it seems a limited workload to bring this feature. We will keep you posted.

**Comment by jbrinkman** [comment]
Just keep in mind that the Workspace Id is not easily discoverable.  Workspace name is much more use friendly.

**Comment by thim81** [comment]
So you suggest to use the CLI option 'postmanWorkspaceName"?

**Comment by jbrinkman** [comment]
Yes.  The workspace Id is hidden in postman and the only way to access it is by calling the Postman Workspaces API.  So the postmanWorkspaceName is a much more friendly approach.  This would behave similar to how Collections are created today where the system looks for the postmanUid first and if it isn't available, then falls back to using the CollectionName.

**Comment by thim81** [comment]
@jbrinkman We gave the Workspace support a try.

There is one big blocker right now for us to move forward.
You can define the workspace ID when creating a collection, in the docs there is no method that allows us to filter collections per workspace.

While doing the investigation, could you explain us your use-case?
1) If the use-case is where you  define the Postman UID as CLI option, to update a collection, then the workspace is not needed.
2) If the use-case is where you, dont define the Postman UID, to create a collection based on the name, then the workspace can be relevant. The problem lies that we automatically, verify if a Postman collection with the same name exists, and use that to create OR update the Postman collection (based on the name). This is make it easier for users since the Postman UID is not so straightforward to find + it pins everything to this 1 UID. So if you remove the collection, the UID is invalid, while with the collection name, Portman is smart and will create or update appropriately. 
3) If the use-case is where you, dont define the Postman UID, to update a collection based on the name, then the workspace is not needed.

It is for use-case (2), that we would need a way to filter the list of collections per workspace, which does not seem to exist in the API (but perhaps you have found a way?) and there is no workspace context in the collection data to filter on.

**Comment by thim81** [comment]
@jbrinkman 

There is a way to get the collections per workspace, which was provided thanks to the Postman community forum:  https://community.postman.com/t/postman-api-filter-collections-by-workspace/3336

So I'll continue to investigate this path for bringing workspace support.

**Comment by thim81** [comment]
@jbrinkman The Workspace PR is ready and will be reviewed and will run some more manual tests to make sure all the use-cases remain working. If things stay positive, the workspace support option will land in one of the next releases.
